doc/chapters/filesystem/index.rst

```diff
@@ -0,0 +1,102 @@
+containers and where it picks up the different packages from. 
+
+
+Standard filesystem
+===================
+
+The base filesystem is bind mounted from a directory created by lxc, the
+directory is empty by default and is setup by the ``lxc-softwarecontainer.in``
+script, which is run by lxc init. 
+
+* The script creates several basic directories creating a normal unix
+  environment as defined by Filesystem Hierarchy Standard (FHS). 
+* The filesystem is still empty except for directories. 
+* LXC then mounts some basic directories defined in the
+  ``libsoftwarecontainer/softwarecontainer.conf`` (copied to
+  ``/usr/share/lxc/conf`` on installation). 
+* On top of this, non standard mountpoints are also used to mount for example
+  the gateway directories etc inside the filesystem. These are defined using
+  BindMountFolderInContainer, or mounted by the different gateways.
+
+The basic filesystem looks like this on startup::
+
+	.
+	├── config
+	└── rootfs
+	    ├── bin
+	    ├── dev
+	    ├── etc
+	    │   ├── group
+	    │   ├── machine-id
+	    │   ├── passwd
+	    │   ├── pulse
+	    │   │   └── client.conf
+	    │   └── resolv.conf
+	    ├── gateways
+	    ├── home
+	    ├── lib
+	    ├── lib64
+	    ├── proc
+	    ├── root
+	    ├── sbin
+	    ├── tmp
+	    └── usr
+	       	├── lib
+	       	├── lib64
+	       	├── local
+	       	└── sbin
+
+
+Mountpoints
+===========
+
+On top of the above structure, mount points can be created anywhere inside the
+container filestructure using the BindMountFolderInContainer and gateway
+configurations which can mount separate files inside the container filesystem.
+
+
+Write Buffered Filesystems
+==========================
+
+A write buffer can be enabled on containers where it is known that the
+application may be misbehaving towards the underlying filesystem. This could be
+overly high number of writes, etc leading to an early death of filesystems for
+example. In this case, enabling the write buffer should provide a layer of
+protection for the filesystem by only allowing a final write of whatever is in
+the upper layer when the container is shutting down. 
+
+To enable the write buffer, set the ``enableWriteBuffer`` option in the
+``com.pelagicore.SoftwareContainerAgent.CreateContainer(prefix, config)`` call.
+This is done using the config parameter in specific, for example, this JSON
+config::
+
+    [{
+        "enableWriteBuffer": true
+    }]
+
+The underlying mechanism of the write buffer is the ``OverlayFS`` which mounts a
+``lower``, ``upper`` and ``work`` directory together into a mountpoint. Files
+are written temporarily into the ``work`` directory while being worked on and
+then into the ``upper`` directory. The ``upper`` directory is merged on top of
+the ``lower`` directory. If a file from the ``lower`` filesystem is opened and
+then edited, it will be written in the ``upper`` directory and the user of the
+mountpoint will see only the changed file. The ``upper`` directory is a
```

Comment:
It might be worth mentioning (unless it is too obvious), that the reverse is not true (opening a file in upper/work does not make it show up in lower until we sync).